Source/Initialization/WarpXInitData.cpp

```diff
@@ -747,6 +747,27 @@ WarpX::InitLevelData (int lev, Real /*time*/)
                    E_ext_grid_s.begin(),
                    ::tolower);
 
+    // Externally imposed fields are only initialized until the user-defined maxlevel_extEMfield_init.
+    // The default maxlevel_extEMfield_init value is the total number of levels in the simulation
+    // Note that if the external fields are not provided, the fields, E and B, on all levels
+    // will still be initialized with the default value of 0.
+    if ( (B_ext_grid_s == "constant") || (B_ext_grid_s == "parse_b_ext_grid_function")) {
+        if (lev > maxlevel_extEMfield_init) {
+            Bfield_fp[lev][0]->setVal(0.);
+            Bfield_fp[lev][1]->setVal(0.);
+            Bfield_fp[lev][2]->setVal(0.);
+            return;
+        }
+    }
+    if ( (E_ext_grid_s == "constant") || (E_ext_grid_s == "parse_e_ext_grid_function")) {
+        if (lev > maxlevel_extEMfield_init) {
+            Efield_fp[lev][0]->setVal(0.);
+            Efield_fp[lev][1]->setVal(0.);
+            Efield_fp[lev][2]->setVal(0.);
+            return;
+        }
+    }
```

Comment:
Just a question to double check if I understand what is meant to be done here: is it correct that the code executes `return;` within these `if` blocks, hence skipping the rest of the function, including what possibly needs to be done for the other field?

Reply:
Hey @EZoni !! 
Thank you so much for catching that. I have fixed it now